lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -97,6 +98,30 @@ def add_customer_data(request, options, payment)
         }.compact
       end
 
+      def add_custom_fields_data(request, options)
+        if options[:fingerprint].present?
+          request[:DeviceFingerprint] = options[:fingerprint]
+          request[:ViaAgent] = false
+        end
+        add_shipping_data(request, options) if options[:consumer_taxes].present?
+        request[:RateOfferId] = options[:rate_offer_id] if options[:rate_offer_id].present?
+        request[:Items] = options[:items] if options[:items].present?
+      end
+
+      def add_shipping_data(request, options)
+        request[:Shipping] = {
+          ConsumerPrice: options[:consumer_price],
+          ConsumerTaxes: options[:consumer_taxes],
+          ConsumerDuty: options[:consumer_duty]
```

Comment:
Unless we know there are other specific types of these fields used by this API, `tax` and `duty` are commonly used elsewhere, so I think we can remove the `consumer_` and use these here:
`price`
`tax`
`duty`

Reply:
Noted! corrected.